pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -125,7 +125,7 @@ func serviceHostname(name, namespace, domainSuffix string) model.Hostname {
 
 // canonicalToIstioServiceAccount converts a Canonical service account to an Istio service account
 func canonicalToIstioServiceAccount(saname string) string {
-	return fmt.Sprintf("%v://%v", istioURIPrefix, saname)
+	return fmt.Sprintf("%v://accounts.google.com/%v", istioURIPrefix, saname)
```

Comment:
why the accounts.google.com ? whats the relevance?

Reply:
The original author @lei-tang [suggests](https://github.com/istio/istio/pull/8657#issuecomment-420864129): `As the subject of this PR is cherrypicking two existing commits, you may make a new PR to remove hardcoded google value and commit the new PR after this cherrypicking PR is merged.`

Now I'm not sure whether I should remove the hardcoded google value in this PR, or in another new PR?

Reply:
I am ok with removing the hardcoded google value in this PR, e.g., you can revise your commit 07dd182 "fix wrong variable name" to remove the hardcoded google value.